compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1523,6 +1531,162 @@ toCtor env con =
 isRecordCtor :: Ctor -> Bool
 isRecordCtor (Ctor _ fldNames fldTys) = not (null fldNames) || null fldTys
 
+------------------------------------------------------------------------------
+-- EXTERNAL PACKAGES
+
+-- External instance methods
+convertExternal :: Env -> LF.PackageRef -> String -> LF.Type -> LF.Expr
+convertExternal env stdlibRef primId lfType
+    | [pkgId, modStr, templName, method] <- splitOn ":" primId
+    , Just LF.Template {..} <- lookup pkgId modStr templName =
+        let pkgRef = PRImport $ PackageId $ T.pack pkgId
+            mod = ModuleName $ map T.pack $ splitOn "." modStr
+            qualify tconName =
+                Qualified
+                    { qualPackage = pkgRef
+                    , qualModule = mod
+                    , qualObject = tconName
+                    }
+            templateDataType = TCon . qualify
+         in case method of
+                "signatory" ->
+                    ETmLam
+                        (tplParam, templateDataType tplTypeCon)
+                        tplSignatories
+                "observer" ->
+                    ETmLam (tplParam, templateDataType tplTypeCon) tplObservers
+                "agreement" ->
+                    ETmLam (tplParam, templateDataType tplTypeCon) tplAgreement
+                "ensure" ->
+                    ETmLam
+                        (tplParam, templateDataType tplTypeCon)
+                        tplPrecondition
+                "create" ->
+                    ETmLam
+                        (tplParam, templateDataType tplTypeCon)
+                        (EUpdate $ UCreate (qualify tplTypeCon) (EVar tplParam))
+                "fetch" ->
+                    let coid = mkVar "$coid"
+                     in ETmLam
+                            ( coid
+                            , TApp
+                                  (TBuiltin BTContractId)
+                                  (templateDataType tplTypeCon))
+                            (EUpdate $ UFetch (qualify tplTypeCon) (EVar coid))
+                "archive" ->
+                    let archiveChoice = ChoiceName "Archive"
+                     in case NM.lookup archiveChoice tplChoices of
+                            Nothing -> errorExpr "archive"
```

Comment:
The error here seems a bit confusing. It sounds like this case is missing in your implementation whereas what’s missing is the `archive` choice in the DALF so we cannot possibly come up with an implementation for this.

Reply:
made the error clearer.